test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -0,0 +1,318 @@
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		framework.Logf("deleting snapshot %q/%q", snapshot.GetNamespace(), snapshot.GetName())
+		// typically this snapshot has already been deleted
+		err = dynamicClient.Resource(snapshotGVR).Namespace(snapshot.GetNamespace()).Delete(snapshot.GetName(), nil)
+		if err != nil && !apierrs.IsNotFound(err) {
+			framework.Failf("Error deleting snapshot %q. Error: %v", claim.Name, err)
+		}
+	}()
+	err = WaitForSnapshotReady(dynamicClient, snapshot.GetNamespace(), snapshot.GetName(), framework.Poll, framework.SnapshotCreateTimeout)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("checking the snapshot")
+	// Get new copy of the snapshot
+	snapshot, err = dynamicClient.Resource(snapshotGVR).Namespace(snapshot.GetNamespace()).Get(snapshot.GetName(), metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+
+	// Get the bound snapshotContent
+	snapshotSpec := snapshot.Object["spec"].(map[string]interface{})
+	snapshotContentName := snapshotSpec["snapshotContentName"].(string)
+	snapshotContent, err := dynamicClient.Resource(snapshotContentGVR).Get(snapshotContentName, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+
+	snapshotContentSpec := snapshotContent.Object["spec"].(map[string]interface{})
+	volumeSnapshotRef := snapshotContentSpec["volumeSnapshotRef"].(map[string]interface{})
+	persistentVolumeRef := snapshotContentSpec["persistentVolumeRef"].(map[string]interface{})
+
+	// Check SnapshotContent properties
+	By("checking the SnapshotContent")
+	Expect(snapshotContentSpec["snapshotClassName"]).To(Equal(snapshotClass.GetName()))
+	Expect(volumeSnapshotRef["name"]).To(Equal(snapshot.GetName()))
+	Expect(volumeSnapshotRef["namespace"]).To(Equal(snapshot.GetNamespace()))
+	Expect(persistentVolumeRef["name"]).To(Equal(pv.Name))
+
+	// Run the checker
+	if t.SnapshotContentCheck != nil {
+		err = t.SnapshotContentCheck(snapshotContent)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	return snapshotContent
+}
+
+// WaitForSnapshotReady waits for a VolumeSnapshot to be ready to use or until timeout occurs, whichever comes first.
+func WaitForSnapshotReady(c dynamic.Interface, ns string, snapshotName string, Poll, timeout time.Duration) error {
+	framework.Logf("Waiting up to %v for VolumeSnapshot %s to become ready", timeout, snapshotName)
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(Poll) {
+		snapshot, err := c.Resource(snapshotGVR).Namespace(ns).Get(snapshotName, metav1.GetOptions{})
+		framework.Logf("get snapshot: %v", snapshot)
+		if err != nil {
+			framework.Logf("Failed to get claim %q, retrying in %v. Error: %v", snapshotName, Poll, err)
+			continue
+		} else {
+			status := snapshot.Object["status"]
+			if status == nil {
+				framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
+				continue
+			}
+			value := status.(map[string]interface{})
+			if value["readyToUse"] == true {
```

Comment:
This field is changed to "readyToUse" in v1.0.1 from "ready" in v0.4.1.  So it won't work with the v0.4.1 snapshotter image and hostpath/hostpath-v0.  See Breaking Changes section in the release notes: https://github.com/kubernetes-csi/external-snapshotter/releases/tag/v1.0.1

Reply:
updated